week1/homework/src/server.js

```diff
@@ -5,11 +5,41 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
+
 function createServer(port) {
   let state = 10;
+  function handleRequest(response, fileType) {
+    response.writeHead(200, { 'Content-Type': fileType });
+    response.write(JSON.stringify({ state }));
+  }
+
+  function handleError(response, fileType) {
+    response.writeHead(404, { 'Content-Type': fileType });
+    response.write(JSON.stringify({ error: 'Not found' }));
+  }
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        handleRequest(response, 'application/json');
+        break;
+      case '/add':
+        state++;
```

Comment:
Great to see you separated the logic of the application i.e increment state from the technical details such as setting head ers and status codes.